app/src/features/Mappings/Edit/EditMapping.tsx

```diff
@@ -0,0 +1,334 @@
+
+            if (!prevJoin) {
+              // Join is created
+              return createJoin({
+                joinRequest: {
+                  column: createdOrUpdatedColumns[index].id,
+                } as JoinRequest,
+              }).unwrap();
+            } else if (!isEqual(prevJoin, join)) {
+              // Join is updated
+              return updateJoin({
+                id: prevJoin.id,
+                joinRequest: { ...prevJoin, ...join },
+              }).unwrap();
+            } else {
+              // Join is unchanged
+              return join;
+            }
+          })
+        );
+
+        // Join columns creation/update
+        await Promise.all(
+          columnsWithJoin.map((column) => {
+            const prevColumn = prevColumnsWithJoin.find(
+              ({ id }) => id === column.id
+            );
+            const index = joins.findIndex((join) => join.id === column.join);
+
+            if (!prevColumn) {
+              // Column is created
+              return createColumn({
+                columnRequest: {
+                  ...column,
+                  join: createdOrUpdatedJoins[index].id,
+                } as ColumnRequest,
+              }).unwrap();
+            } else if (!isEqual(prevColumn, column)) {
+              // Column is updated
+              return updateColumn({
+                id: prevColumn.id,
+                columnRequest: {
+                  ...prevColumn,
+                  ...column,
+                },
+              }).unwrap();
+            } else {
+              // Column is unchanged
+              return column;
+            }
+          })
+        );
+
+        // Delete order is inverted from creation order
+        const deletedColumnsWithoutJoin = differenceBy(
+          prevColumnsWithoutJoin,
+          columnsWithoutJoin,
+          ({ id }) => id
+        );
+        const deletedFilters = differenceBy(
+          initialState.filters,
+          filters,
+          ({ id }) => id
+        );
+        const deletedJoins = differenceBy(
+          initialState.joins,
+          joins,
+          ({ id }) => id
+        );
+        const deletedColumnsWithJoin = differenceBy(
+          prevColumnsWithJoin,
+          columnsWithJoin,
+          ({ id }) => id
+        );
+        await Promise.all(
+          deletedColumnsWithJoin.map((column) =>
+            deleteColumn({ id: column.id }).unwrap()
+          )
+        );
+        await Promise.all(
+          deletedJoins.map((join) => deleteJoin({ id: join.id }).unwrap())
+        );
+        await Promise.all(
+          deletedFilters.map((filter) =>
+            deleteFilter({ id: filter.id }).unwrap()
+          )
+        );
+        await Promise.all(
+          deletedColumnsWithoutJoin.map((column) =>
+            deleteColumn({ id: column.id }).unwrap()
+          )
+        );
+      } catch (error) {
+        // Fix: Handle Column, Filter, Join creation/update/delete errors
```

Comment:
how do you intend to do this ?

Reply:
Hmm dunno.. We just need to agree on the behavior when this happens

Maybe showing an alert and preventing the route change ?

Reply:
@BPierrick  I guess using an error-snackbar would be fine for now maybe : https://material-ui.com/components/snackbars/

Reply:
The issue is knowing what to show in the snackbar because there's no way here to tell what specifically failed

Reply:
Doesn't django returns formatted errors like you did in `CredentialForm` ? 

```typescript
  catch (e) {
      const data = apiValidationErrorFromResponse<Partial<CredentialRequest>>(
        e as FetchBaseQueryError
      );
      setAlert(head(data?.non_field_errors));
    }
```

Reply:
no because it won't be a non field error in those cases